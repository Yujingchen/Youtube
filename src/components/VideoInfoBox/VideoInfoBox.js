import React, { Component } from "react";
import "./VideoInfoBox.scss";
import { Image, Button, Divider } from "semantic-ui-react";

class VideoInfoBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      collapsed: true
    };
  }

  onToggleCollapseButtonClick = () => {
    this.setState(prevState => {
      return {
        collapsed: !prevState.collapsed
      };
    });
  };

  render() {
    let descriptionTextClass = "collapsed";
    let buttonTitle = "Show More";
    if (!this.state.collapsed) {
      descriptionTextClass = "expanded";
      buttonTitle = "Show Less";
    }

    return (
      <div>
        <div className="video-info-box">
          <Image
            className="channel-image"
            src="http://via.placeholder.com/48x48"
            circular
          />
          <div className="video-info">
            <div className="channel-name"> Channel Name</div>
            <div className="video-publication-date">Thu 24, 2017</div>
          </div>

          <Button color="youtube" className="subscribe">
            91.5K Subscribe
          </Button>

          <div className="video-description">
            <div className={descriptionTextClass}>
              <p>p1</p>
              <p>p2</p>
              <p>p3</p>
              <p>p4</p>
              <p>p5</p>
            </div>
            <Button onClick={this.onToggleCollapseButtonClick} compact>
              {buttonTitle}
            </Button>
          </div>
        </div>
        <Divider />
      </div>
    );
  }
}

export default VideoInfoBox;
