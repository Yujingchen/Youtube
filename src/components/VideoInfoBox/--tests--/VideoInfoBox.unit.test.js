import React from "react";
import VideoInfoBox from "../VideoInfoBox";
import { shallow } from "enzyme";
describe("VideoInfobox", () => {
  test("renders collapsed", () => {
    const wrapper = shallow(<VideoInfoBox />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders expanded", () => {
    const wrapper = shallow(<VideoInfoBox />);
    wrapper.setState({ collapsed: false });
    expect(wrapper).toMatchSnapshot();
  });
});
