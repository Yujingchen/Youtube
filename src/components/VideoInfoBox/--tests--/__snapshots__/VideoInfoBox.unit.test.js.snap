// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInfobox renders collapsed 1`] = `
<div>
  <div
    className="video-info-box"
  >
    <Image
      as="img"
      circular={true}
      className="channel-image"
      src="http://via.placeholder.com/48x48"
      ui={true}
    />
    <div
      className="video-info"
    >
      <div
        className="channel-name"
      >
         Channel Name
      </div>
      <div
        className="video-publication-date"
      >
        Thu 24, 2017
      </div>
    </div>
    <Button
      as="button"
      className="subscribe"
      color="youtube"
    >
      91.5K Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="collapsed"
      >
        <p>
          p1
        </p>
        <p>
          p2
        </p>
        <p>
          p3
        </p>
        <p>
          p4
        </p>
        <p>
          p5
        </p>
      </div>
      <Button
        as="button"
        compact={true}
        onClick={[Function]}
      >
        Show More
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`VideoInfobox renders expanded 1`] = `
<div>
  <div
    className="video-info-box"
  >
    <Image
      as="img"
      circular={true}
      className="channel-image"
      src="http://via.placeholder.com/48x48"
      ui={true}
    />
    <div
      className="video-info"
    >
      <div
        className="channel-name"
      >
         Channel Name
      </div>
      <div
        className="video-publication-date"
      >
        Thu 24, 2017
      </div>
    </div>
    <Button
      as="button"
      className="subscribe"
      color="youtube"
    >
      91.5K Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          p1
        </p>
        <p>
          p2
        </p>
        <p>
          p3
        </p>
        <p>
          p4
        </p>
        <p>
          p5
        </p>
      </div>
      <Button
        as="button"
        compact={true}
        onClick={[Function]}
      >
        Show Less
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;
